pkg/helper/download/download.go

```diff
@@ -0,0 +1,82 @@
+package helper
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/dustin/go-humanize"
+)
+
+// download will download a url to a local file. It's efficient because it will
+// write as it downloads and not load the whole file into memory. We pass an io.TeeReader
+// into Copy() to report progress on the download.
+func Download(url string, filename string) error {
+	// Create the file, but give it a tmp file extension, this means we won't overwrite a
+	// file until it's downloaded, but we'll remove the tmp extension once downloaded.
+	out, err := os.Create(filename + ".tmp")
+	if err != nil {
+		return err
+	}
+
+	// Get the data
+	// nolint:gosec
+	resp, err := http.Get(url)
+	if err != nil {
+		out.Close()
+		return err
+	}
+	defer resp.Body.Close()
+
+	// Create our progress reporter and pass it to be used alongside our writer
+	counter := &WriteCounter{}
+	if _, err = io.Copy(out, io.TeeReader(resp.Body, counter)); err != nil {
+		out.Close()
+		return err
+	}
+
+	// The progress use the same line so print a new line once it's finished downloading
+	fmt.Print("\n")
+
+	// Close the file without defer so it can happen before Rename()
+	out.Close()
+
+	if err = os.Rename(filename+".tmp", filename); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Get the extension used for the compressed oc file
+func GetExtension() string {
+	if runtime.GOOS == "windows" {
+		return "zip"
+	}
+	return "tar.gz"
+}
+
+// WriteCounter counts the number of bytes written to it. It implements to the io.Writer interface
+// and we can pass this into io.TeeReader() which will report progress on each write cycle.
+type WriteCounter struct {
```

Comment:
NP
We can have this new struct in the helper files since it can be used in other commands in the future.